test/e2e_node/runner/run_e2e.go

```diff
@@ -122,41 +137,102 @@ func main() {
 	}
 }
 
-func WaitForUser() {
-	scanner := bufio.NewScanner(os.Stdin)
-	fmt.Printf("Enter \"y\" to run tests\n")
-	for scanner.Scan() {
-		if strings.ToUpper(scanner.Text()) != "Y\n" {
-			break
+func testHost(host string) *TestResult {
+	return RunTestsAgainstHost(*project, *zone, host, kubeRoot)
+}
+
+func testImage(image string) *TestResult {
+	err := createInstance(image)
+	defer deleteInstance(image)
+	if err != nil {
+		return &TestResult{
+			Err: fmt.Errorf("Unable to create gce instance for image %s.  %v", image, err),
 		}
-		fmt.Printf("Enter \"y\" to run tests\n")
 	}
-	u.Done()
+	result := RunTestsAgainstHost(*project, *zone, imageToInstanceName(image), kubeRoot)
+	return result
 }
 
-type TestResult struct {
-	fullhost                string
-	err                     error
-	testCombinedOutput      string
-	etcdCombinedOutput      string
-	apiServerCombinedOutput string
-	kubeletCombinedOutput   string
+func createInstance(image string) error {
+	name := imageToInstanceName(image)
+	i := &compute.Instance{
+		Name:        name,
+		MachineType: machineType(),
+		NetworkInterfaces: []*compute.NetworkInterface{
+			{
+				AccessConfigs: []*compute.AccessConfig{
+					{
+						Type: "ONE_TO_ONE_NAT",
+						Name: "External NAT",
+					},
+				}},
+		},
+		Disks: []*compute.AttachedDisk{
+			{
+				AutoDelete: true,
+				Boot:       true,
+				Type:       "PERSISTENT",
+				InitializeParams: &compute.AttachedDiskInitializeParams{
+					SourceImage: sourceImage(image),
+				},
+			},
+		},
+	}
+	_, err := computeService.Instances.Insert(*project, *zone, i).Do()
+	if err != nil {
+		return err
+	}
+
+	for i := 0; i < 10; i++ {
+		instance, err := computeService.Instances.Get(*project, *zone, name).Do()
+		if err == nil {
+			if strings.ToUpper(instance.Status) == "RUNNING" {
+				glog.Infof("Instance %s RUNNING", name)
+				break
+			}
+			glog.Infof("Waiting for instance %s to be RUNNING  was %s", name, instance.Status)
+		}
+		time.Sleep(time.Second * 30)
+	}
+	// Sleep an additional minute to let the vm be ready
+	time.Sleep(time.Minute)
```

Comment:
Could this be flaky? Can we perform some health check instead like `[ -n ssh -- docker version | grep Server ]` or something similar?